crates/bifrost/src/providers/replicated_loglet/sequencer/mod.rs

```diff
@@ -297,7 +168,36 @@ impl<T: TransportConnect> Sequencer<T> {
             return Ok(LogletCommit::sealed());
         }
 
-        let next_write_offset = records.last_offset(self.next_write_offset)?.next();
+        if payloads.len() > self.max_in_flight_records_in_config {
+            let delta = payloads.len() - self.max_in_flight_records_in_config;
+            debug!(
+                "Resizing sequencer in-flight records capacity to allow admission for this batch. \
+                Capacity in configuration is {} and we are adding capacity of {} to it",
+                self.max_in_flight_records_in_config, delta
+            );
+            self.record_permits.add_permits(delta);
+        }
+
+        let len = u32::try_from(payloads.len()).expect("batch sizes fit in u32");
+        let permit = self
+            .record_permits
+            .clone()
+            .acquire_many_owned(len)
+            .await
+            .unwrap();
+
+        // Why is this AclRel?
+        // We are updating the next write offset and we want to make sure that after this call that
+        // we observe if task_center()'s shutdown signal was set or not consistently across
+        // threads.
+        //
+        // The situation we want to avoid is that we fail to spawn an appender due to shutdown but
+        // the subsequent fetch_add don't observe task-center's internal shutdown atomic.
+        let offset = LogletOffset::new(
+            self.sequencer_shared_state
+                .next_write_offset
+                .fetch_add(len, std::sync::atomic::Ordering::AcqRel),
+        );
```

Comment:
I am not 100% sure that this atomic operation does what the comment describes. Couldn't the following happen: Thread 1 calls `enqueue_batch` and obtains `offset == 1`. Thread 2 calls `enqueue_batch` and obtains `offset == 2`. Thread 2 continues and spawns the `appender`. Now, thread 3 shuts the task center down. Then thread 1 continues its work and tries to spawn an appender. Since thread 3 shut the task center down, this operation fails. Wouldn't this scenario be isomorphic to one `enqueue_batch` fails to spawn the appender but subsequent ones succeed?

Wouldn't `Ordering::Acquire` be enough to prevent the shutdown atomic from being reordered before?